statistics/histogram.go

```diff
@@ -967,6 +990,29 @@ func (idx *Index) outOfRange(val types.Datum) bool {
 	return !withInLowBoundOrPrefixMatch || !withInHighBound
 }
 
+// RemoveUpperBound removes the upper bound the index stats.
+// It is used when merge stats for incremental analyze.
+func (idx *Index) RemoveUpperBound(sc *stmtctx.StatementContext, values []types.Datum) (*Histogram, *CMSketch, error) {
+	hist, cms := idx.Histogram.Copy(), idx.CMSketch.Copy()
+	buckets := hist.Buckets
```

Comment:
What the usage for this copy?

Reply:
To avoid data race.

Reply:
Why existing data race? The histogram has been copied.